Feature/testcases

samuelnoye opened the pull request.

samilaxy committed:
update

samilaxy committed:
update

samilaxy committed:
update

samilaxy committed:
Update UserInterfaceState.xcuserstate

FitzAfful commented on LoggerFrameworkProjectmainTests/LoggerTest.swift:
```diff
@@ -0,0 +1,38 @@
+    //create SUT
+    
+    var sut: Logger!
+    var env = Logger.Environment.self
```
env isnt used anywhere in your tests

FitzAfful commented on LoggerFrameworkProjectmainTests/LoggerTest.swift:
```diff
@@ -0,0 +1,38 @@
+        var result = sut.log("dgg", severity: .veryLow)
+        //Then
+        XCTAssertEqual(result, logMessage)
+    }
```
The method you're testing is called log so using a test method name of testLogger doesn't show what exactly you're testing. A more appropriate name would be 'testLog'

* Please leave a line before your //Given, //When and //Then

FitzAfful commented on LoggerFrameworkProjectmainTests/LoggerTest.swift:
```diff
@@ -0,0 +1,38 @@
+        sut = nil
+    }
+
+    func testLogger(){
+        //Given
+        
+        sut.log("dgg", severity: .veryLow)
+        var logMessage = sut.logMessage
+        //When
+        var result = sut.log("dgg", severity: .veryLow)
+        //Then
+        XCTAssertEqual(result, logMessage)
+    }
```
It is also important to test literals. 

So 
        //Given
        sut.log("dgg", severity: .veryLow)
        
        //When
        var result = sut.log("dgg", severity: .veryLow)
        
        //Then
        XCTAssertEqual(result, "The exact string that result should be")
        XCTAssertEqual(sut.logMessage, "The exact string that result should be")

samilaxy replied:
Okay

FitzAfful commented on LoggerFrameworkProjectmain/Logger.swift:
```diff
@@ -44,6 +45,7 @@ class Logger {
         }
+        return logMessage
     }
     func veryHigh(_ message: String){
```
I didnt see tests for all of these. Please share your code coverage when updating or creating a PR.

samilaxy replied:
Okay, sure

FitzAfful commented:
I asked Maxwell to share his comments with you. Please do same with him. Especially about commit messages etc.

samilaxy committed:
test literals.

samilaxy committed:
remove env

samilaxy committed:
Update test logger

-rename logger test function
-get current timestamp

samilaxy committed:
Update severity function

-switch case for severity
-create testSeverity function
-test all cases for severity

samilaxy committed:
UserDefault

samilaxy committed:
Update logger class

- saveLogToFile
- readFromFile
